bundle/src/main/java/com/adobe/acs/commons/replication/dispatcher/refetchflush/impl/RefetchFlushContentBuilderImpl.java

```diff
@@ -0,0 +1,421 @@
+                PropertiesUtil.toStringArray(properties.get(PROP_EXTENSION_PAIRS),
+                        DEFAULT_EXTENSION_PAIRS), "=", false, null, false));
+
+        logInfoMessage("Extension Pairs [" +  mapToString(this.extensionPairs) + "]");
+
+        ArrayList<String> validMatches = new ArrayList<>();
+        String[] matchProps = PropertiesUtil.toStringArray(properties.get(PROP_MATCH_PATH), DEFAULT_MATCH_PATH);
+        if (matchProps.length > 0) {
+            for (String match : matchProps) {
+                if (StringUtils.isNotEmpty(match)) {
+                    validMatches.add(match);
+                }
+            }
+        }
+
+        this.pathMatches = validMatches.toArray(new String[0]);
+
+        logInfoMessage("Match Path Patterns [" +  String.join(",", this.pathMatches) + "]");
+    }
+
+    @SuppressWarnings("unused")
+    @Deactivate
+    protected final void deactivate(final Map<String, Object> properties) {
+        this.extensionPairs = new HashMap<>();
+        this.pathMatches = DEFAULT_MATCH_PATH;
+    }
+
+    /**
+     * Take the mapped extensions and organize them by individual extension.
+     * @param configuredExtensions Map of extension mappings
+     * @return Map with extension as keys
+     */
+    private Map<String, String[]> formatExtensions(final Map<String, String> configuredExtensions) {
+        final Map<String, String[]> extensions = new LinkedHashMap<>();
+
+        for (final Map.Entry<String, String> entry : configuredExtensions.entrySet()) {
+            final String ext = entry.getKey().trim();
+            extensions.put(ext, entry.getValue().trim().split("&"));
+        }
+
+        return extensions;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ReplicationContent create(Session session, ReplicationAction replicationAction,
+                                     ReplicationContentFactory factory, Map<String, Object> options)
+            throws ReplicationException {
+        return create(session, replicationAction, factory);
+    }
+
+    public ReplicationContent create(Session session, ReplicationAction action, ReplicationContentFactory factory)
+            throws ReplicationException {
+        String path = action.getPath();
+        replicationLog = action.getLog();
+        if (replicationLog == null) {
+            logWarnMessage("No replication log found on agent " + NAME);
+        }
+
+        /* Check if the action is valid, and whether it should be ignored. */
+        checkValidity(action, path);
+        if (shouldIgnore(action, path)) {
+            return ReplicationContent.VOID;
+        }
+
+        logInfoMessage("Content builder invoked for path " + path + ", with replication action "
+                + action.getType() + " and serialization type " +  action.getConfig().getSerializationType());
+
+        String[] uris;
+        int extSep = path.indexOf('.', path.lastIndexOf('/'));
+        if (extSep == -1) {
```

Comment:
I thought about other additions here, to help make the activation more powerful and flexible for the user, but thought I'd smart small.